src/Microsoft.AspNetCore.Mvc.TagHelpers/InputTagHelper.cs

```diff
@@ -230,6 +230,33 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
             }
         }
 
+        /// <summary>
+        /// Gets an &lt;input&gt; elements "type" attribute value for the given <paramref name="modelExplorer"/>.
+        /// </summary>
+        /// <param name="modelExplorer">The <see cref="ModelExplorer"/> to inspect.</param>
+        /// <param name="inputTypeHint">When this method returns, contains a type hint associated with the given
+        /// <paramref name="modelExplorer"/>.</param>
+        /// <returns>An &lt;input&gt; elements "type" attribute value.</returns>
+        /// <remarks><paramref name="inputTypeHint"/> values are based off of the <paramref name="modelExplorer"/>'s
+        /// <see cref="ModelExplorer.Metadata"/>. Input type hints are typically used to format an
+        /// &lt;input&gt; elements "value" attribute of a generated text box or used to determine if a text box should
+        /// be multi-line.</remarks>
+        protected string GetInputType(ModelExplorer modelExplorer, out string inputTypeHint)
```

Comment:
For the future, please do not move methods around while making other changes. It makes it harder to see what's really changing. We also aren't that persnickety about order in existing classes and someone could do a pass reordering class members on some future Engineering Day.